src/crimson/os/seastore/onode_manager/staged-fltree/tree.cc

```diff
@@ -0,0 +1,231 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "tree.h"
+
+#include "node.h"
+#include "node_extent_manager.h"
+#include "stages/key_layout.h"
+#include "super.h"
+
+namespace crimson::os::seastore::onode {
+
+using btree_ertr = Btree::btree_ertr;
+template <class ValueT=void>
+using btree_future = Btree::btree_future<ValueT>;
+using Cursor = Btree::Cursor;
+
+Cursor::Cursor(Btree* p_tree, Ref<tree_cursor_t> _p_cursor)
+  : p_tree(p_tree) {
+  if (_p_cursor->is_end()) {
+    // no need to hold the leaf node
+  } else {
+    p_cursor = _p_cursor;
+  }
+}
+Cursor::Cursor(Btree* p_tree) : p_tree{p_tree} {}
+Cursor::Cursor(const Cursor&) = default;
+Cursor::Cursor(Cursor&&) noexcept = default;
+Cursor& Cursor::operator=(const Cursor&) = default;
+Cursor& Cursor::operator=(Cursor&&) = default;
+Cursor::~Cursor() = default;
+
+bool Cursor::is_end() const {
+  if (p_cursor) {
+    assert(!p_cursor->is_end());
+    return false;
+  } else {
+    return true;
+  }
+}
+
+ghobject_t Cursor::get_ghobj() const {
+  return p_cursor->get_key_view().to_ghobj();
+}
+
+const onode_t* Cursor::value() const {
+  return p_cursor->get_p_value();
+}
+
+bool Cursor::operator==(const Cursor& x) const {
+  return p_cursor == x.p_cursor;
+}
+
+Cursor& Cursor::operator++() {
+  // TODO
```

Comment:
?

Reply:
Not yet for iteration, currently only insert, lookup and split are implemented.